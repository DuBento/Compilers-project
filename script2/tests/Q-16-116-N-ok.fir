void *fill(<int> unsorted) {
  unsorted[0] = 83;
  unsorted[1] = 62;
  unsorted[2] = 81;
  unsorted[3] = 26;
  unsorted[4] = 39;
  unsorted[5] = 97;
  unsorted[6] = 59;
  unsorted[7] = 56;
  unsorted[8] = 86;
  unsorted[9] = 21;
}

void *swap(<int> unsorted, int a, int b) {
  int temp = unsorted[a];
  unsorted[a] = unsorted[b];
  unsorted[b] = temp;
}

void *sort(<int> unsorted) {
  int i = 0;
  while (i < 9) do {
    int smallIndex = i;
    int j = i + 1;

    while (j < 10) do {
      if (unsorted[j] < unsorted[smallIndex]) then
        smallIndex = j;
      j = j + 1;
    }

    swap(unsorted, i, smallIndex);
    i = i + 1;
  }
}

void *print(<int> sorted) {
  int i = 0;
  while (i < 10) do {
    writeln sorted[i];
    i = i + 1;
  }
}

int *fir() -> 0 @ {
  <int> unsorted = [10];
  fill(unsorted);
} 
{
  sort(unsorted);
}
>> {
  print(unsorted);
}