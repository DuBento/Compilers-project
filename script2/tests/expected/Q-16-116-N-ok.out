21
26
39
56
59
62
81
83
86
97
